src/responder/pam/pamsrv_p11.c

```diff
@@ -424,6 +427,33 @@ struct tevent_req *pam_check_cert_send(TALLOC_CTX *mem_ctx,
         extra_args[arg_c++] = verify_opts;
         extra_args[arg_c++] = "--verify";
     }
+
+    if (pd->authtok != NULL
```

Comment:
@sumit-bose,
may I know why did you add this check?

Because `struct pam_data` are created only on single place and authtok is properly initialized there.
```
sh$ git grep "struct pam_data" | grep talloc
src/providers/dp_pam_data_util.c:    struct pam_data *pd = talloc_get_type(ptr, struct pam_data);
src/providers/dp_pam_data_util.c:    pd = talloc_zero(mem_ctx, struct pam_data);
src/providers/krb5/krb5_delayed_online_authentication.c:        pd = talloc_get_type(entry->value.ptr, struct pam_data);
src/tests/cmocka/data_provider/test_dp_builtin.c:    pd = talloc_zero(test_ctx, struct pam_data);
src/tests/cmocka/data_provider/test_dp_builtin.c:    pd = talloc_zero(test_ctx, struct pam_data);
src/tests/cmocka/test_krb5_wait_queue.c:    test_ctx->pd = talloc_zero(test_ctx, struct pam_data);
src/tests/cmocka/test_pam_srv.c:    pd = talloc_zero(pam_test_ctx, struct pam_data);
src/tests/cmocka/test_simple_access.c:    simple_test_ctx->pd = talloc_zero(simple_test_ctx, struct pam_data);
```

https://pagure.io/SSSD/sssd/blob/master/f/src/providers/dp_pam_data_util.c#_55

Did you want to add check for `pd != NULL` ? or what was the reason.